provider/openrouter/openrouter.go

```diff
@@ -53,44 +57,54 @@ func (p *Provider) ID() (id string) {
 
 // ModelsList holds a list of models.
 type ModelsList struct {
-	Models []Model `json:"data"`
+	Models []model.MetaInformation `json:"data"`
 }
 
-// Model holds a model.
-type Model struct {
-	// ID holds the model id.
-	ID string `json:"id"`
-	// Name holds the model name.
-	Name string `json:"name"`
+// Models returns which models are available to be queried via this provider.
+func (p *Provider) Models() (models []model.Model, err error) {
+	responseModels, err := p.fetchModels()
+	if err != nil {
+		return nil, err
+	}
 
-	// Pricing holds the pricing information of a model.
-	Pricing Pricing `json:"pricing"`
-}
+	models = make([]model.Model, len(responseModels.Models))
+	for i, model := range responseModels.Models {
+		models[i] = llm.NewModel(p, p.ID()+provider.ProviderModelSeparator+model.ID, &model)
```

Comment:
Found an interesting bug here... You pass the pointer to `model` to the constructor because the model here is a struct. In the next iteration the next model information is copied exactly into that struct... So basically you give every LLM constructor the same pointer cause `model` is never re-allocated. And in the end they all share the meta information from the last model. Try it out ^^.

Reply:
Possible fix:

```go
for i, model := range responseModels.Models {
	information := model // Reallocate the information cause "model" is overwritten in the next iteration.
	models[i] = llm.NewModel(p, p.ID()+provider.ProviderModelSeparator+model.ID, &information)
}

```

Reply:
Good catch